kratos/containers/variable.h

```diff
@@ -325,6 +317,19 @@ class Variable : public VariableData
         return GetValueByIndex(static_cast<TDataType*>(pSource),GetComponentIndex());
     }
 
+    /// Add the variable to the Kratos Registry
+    void Register() const {
+        const std::string variable_path("variables.all."+Name());
+        if (!Registry::HasItem(variable_path)) {
+            Registry::AddItem<VariableType>(variable_path, *this);
+            Registry::AddItem<VariableType>("variables."+Registry::GetCurrentSource()+"."+Name(), *this);
+        } else if(!Registry::GetItem(variable_path).IsSameType(*this)) {
+            KRATOS_ERROR << "Attempting to register " << Name()
+                << " but a variable with the same name and different type already exists"
+                << std::endl;
+        }
+    }
```

Comment:
I am curious, since registering the same variable from multiple apps causes some seg fault at the moment, shoudn't we throw an error if it exists, rather than checking the type?

Reply:
I think that Kratos should be able to support registering the same variable from multiple applications, as long as the type is the same. Using the registry in this way would be a step in that direction, since users of the registry are sure to always get the same thing when retrieving a given variable (something that is not so clear with components if Kratos allows overwriting variables as is happening now).